packages/compass-collection/src/modules/stats.ts

```diff
@@ -112,16 +125,21 @@ export const updateCollectionDetails = (
  *
  * @returns {any} The new state.
  */
-const reducer = (
-  state = getInitialState(),
-  action: AnyAction
-): CollectionStatsObject => {
-  if (action.type === UPDATE_COLLECTION_DETAILS) {
-    return action.stats;
-  } else if (action.type === RESET_COLLECTION_DETAILS) {
-    return getInitialState();
+const reducer = (state = getInitialState(), action: AnyAction): State => {
+  switch (action.type) {
+    case ActionTypes.UpdateCollectionDetails:
+      return {
+        ...state,
+        [action.namespace]: action.stats,
+      };
+    case ActionTypes.ResetCollectionDetails:
+      return {
+        ...state,
+        [action.namespace]: getCollectionStatsInitialState(),
```

Comment:
Do you really need to keep it around as initial state on reset taking into account that in the component you are using `??` to fallback to initial anyway?

Reply:
Good thinking, updated it to clean up old ones and not have any stored default state (uses the `??` fallback in the component always).